model_templates/training/python3_sklearn_binary_hyperparameters/custom.py

```diff
@@ -0,0 +1,171 @@
+import pickle
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from create_pipeline import make_classifier
+from sklearn.preprocessing import label_binarize
+
+
+def fit(
+    X: pd.DataFrame,
+    y: pd.Series,
+    output_dir: str,
+    class_order: Optional[List[str]] = None,
+    row_weights: Optional[np.ndarray] = None,
+    parameters: Optional[dict] = None,
+    **kwargs,
+):
+    """
+    This hook must be implemented with your fitting code, for running drum in the fit mode.
+
+    This hook MUST ALWAYS be implemented for custom training models.
+    For inference models, this hook can stick around unimplemented, and won’t be triggered.
+
+    Parameters
+    ----------
+    X: pd.DataFrame - training data to perform fit on
+    y: pd.Series - target data to perform fit on
+    output_dir: the path to write output. This is the path provided in '--output' parameter of the
+        'drum fit' command.
+    class_order : A two element long list dictating the order of classes which should be used for
+        modeling. Class order will always be passed to fit by DataRobot for classification tasks,
+        and never otherwise. When models predict, they output a likelihood of one class, with a
+        value from 0 to 1. The likelihood of the other class is 1 - this likelihood. Class order
+        dictates that the first element in the list will be the 0 class, and the second will be the
+        1 class.
+    row_weights: An array of non-negative numeric values which can be used to dictate how important
+        a row is. Row weights is only optionally used, and there will be no filtering for which
+        custom models support this. There are two situations when values will be passed into
+        row_weights, during smart downsampling and when weights are explicitly provided by the user
+    parameters: dict
+        A dictionary of parameters defined within the model-metadata.yaml file.
+    kwargs: Added for forwards compatibility
+
+    Returns
+    -------
+    Nothing
+    """
+    # Feel free to delete which ever one of these you aren't using
+    if class_order is not None:
+        if y.dtype == np.dtype("bool"):
+            y = y.astype("str")
+        estimator = make_classifier(X)
+    else:
+        raise Exception("Running binary training: class_order expected to be not None")
+
+    if not parameters:
+        raise ValueError("Did not receive parameters")
+
+    # Parameters are provided during fit as a dict with names set according to your model-metadata.yaml file.
+    # In this example, we set the 'model' parameters in our sklearn pipeline.
+    estimator.set_params(
+        **{
+            "model__penalty": parameters["penalty"],
+            "model__dual": parameters["dual"],
```

Comment:
what would you think about just replacing the normal sklearn binary one with this?

Reply:
I'd be okay with that, but first lets get it working e2e in DR